server/src/internalClusterTest/java/org/elasticsearch/action/admin/indices/create/AutoCreateSystemIndexIT.java

```diff
@@ -84,6 +99,24 @@ public void testAutoCreateNonPrimaryIndex() throws Exception {
 
         GetIndexResponse response = client().admin().indices().prepareGetIndex().addIndices(INDEX_NAME + "-2").get();
         assertThat(response.indices().length, is(1));
+        assertThat(response.aliases().size(), is(1));
+        assertThat(response.aliases().get(INDEX_NAME + "-2").size(), is(1));
+        assertThat(response.aliases().get(INDEX_NAME + "-2").get(0), equalTo(AliasMetadata.builder(INDEX_NAME).isHidden(true).build()));
+    }
+
+    public void testAutoCreateSystemIndicesAllowsWriteToAlias() throws Exception {
```

Comment:
Would it be possible to test the opposite scenario? Such that, the first API request makes the non primary index INDEX_NAME-2 and then we add the primary name - 1.